api/src/lib/utils.js

```diff
@@ -106,17 +106,21 @@ module.exports = class Utils {
   * hostLookup (host) {
     let response
     try {
-      response = yield this.webfingerLookup(host)
+      response = yield this._webfingerLookup(host)
     } catch (e) {
       throw new NotFoundError('Host is unavailable')
     }
 
     if (!response) throw new NotFoundError('Host is unavailable')
     if (!response.properties) throw new NotFoundError("Host doesn't have an ilp-kit or the version is not compatible")
 
-    return {
-      publicKey: response.properties['https://interledger.org/rel/publicKey'],
-      peersRpcUri: _.filter(response.links, {rel: 'https://interledger.org/rel/peersRpcUri'})[0].href
+    try {
+      return {
+        publicKey: response.properties['https://interledger.org/rel/publicKey'],
+        peersRpcUri: _.filter(response.links, {rel: 'https://interledger.org/rel/peersRpcUri'})[0].href
+      }
+    } catch (e) {
+      throw new NotFoundError('Host webfinger parsing failed')
```

Comment:
how come? `response.links` should be an array `_.filter` should return an array, `href` should be in the object, otherwise the webfinger result is malformed and an error will be thrown.